cmd/esc/cli/login.go

```diff
@@ -157,7 +154,28 @@ func (esc *escCommand) getCachedClient(ctx context.Context) error {
 		return fmt.Errorf("could not determine current cloud: %w", err)
 	}
 	if account == nil {
-		return fmt.Errorf("no credentials. Please run `%v login` to log in.", esc.command)
+		backendURL := esc.workspace.GetCurrentCloudURL(nil)
+
+		nAccount, err := esc.login.Login(
+			ctx,
+			backendURL,
+			false,
+			"esc",
+			"Pulumi ESC environments",
+			nil,
+			false,
+			display.Options{Color: esc.colors},
+		)
+
+		if err != nil {
+			return fmt.Errorf("could not determine current cloud: %w", err)
+		}
+
+		account = &workspace.Account{
+			Account:    *nAccount,
+			BackendURL: "https://api.pulumi.com",
+			DefaultOrg: "",
```

Comment:
we probably want to populate the default org here as well (see `workspace.GetBackendConfigDefaultOrg`)

Reply:
done